### DISCUSSION TITLE
Error in giving upper bound for discrete_design_range variable

### DISCUSSION THREAD

**Discussion by MuhammadKamranButt** [snl-dakota/dakota#97] [Q&A] [answered]
Hello, I am using DAKOTA for optimization problems and using the MOGA package. When I define upper bound of discrete_design_range variable to 20922789888000 (16 factorial), it gives me error. The error says "lower bound for discrete_design_range variable x1 exceeds upper bound. It also gives me warning, " Setting initial_value for discrete_design_range variable x1 to upper bound. What is the upper bound limit in DAKOTA?

**Comment by jadamstephens** [comment, marked as answer]
Discrete design range variables are stored as ordinary integers. On most systems that's four bytes, and INT_MAX is 2147483647. The error message regrettably is not clear.

A couple of possible workarounds:

* If the range of the variable is very large, use a continuous_design variable and round/floor it to an int in your driver, if the language your driver and simulation are written in can handle integers that big. If you run the optimizer long enough it'll eventually get confused because it can't tell the difference between, say 20922789888000.1 and 20922789888000.2, but that's probably not a practical concern unless the range of the variable is small.
* If the range of the variable is small, offset the variable in Dakota by setting the lower bound to 0 and upper bound to the range. In your driver, add the offset back. Again, this assumes that your driver language and simulation can take these large integers.

**Reply by MuhammadKamranButt** [reply]
Sir, thank you very much. Following your suggestion, I changed the variable to "continuous_design" and rounded it in my Python analysis_driver script. It's working well now. I truly appreciate your help.
